.github/workflows/project-checks.yaml

```diff
@@ -0,0 +1,55 @@
+name: Project Checks
+
+on:
+  push:
+    branches:
+      - main
+      - "release/**"
+      - "test/**"
+  pull_request:
+    branches:
+      - main
+      - "release/**"
+
+env:
+  GO_VERSION: "1.19.5"
+
+jobs:
+  checkout:
+    name: Checkout
+    runs-on: ubuntu-22.04
+    timeout-minutes: 5
+    steps:
+      - name: Set env
+        shell: bash
+        run: |
+          echo "GOPATH=${{ github.workspace }}" >> $GITHUB_ENV
+          echo "${{ github.workspace }}/bin" >> $GITHUB_PATH
+      - uses: actions/checkout@v3
+        with:
+          path: src/github.com/containerd/nri-plugin-collection
+          fetch-depth: 25
+
+  verify:
+    name: Verify
+    needs: [checkout]
+    runs-on: ubuntu-22.04
+    timeout-minutes: 5
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - uses: actions/checkout@v3
+        with:
+          path: src/github.com/containerd/nri-plugin-collection
+          fetch-depth: 25
+      - name: Set Up Environment
+        shell: bash
+        run: |
+          echo "GOPATH=${{ github.workspace }}" >> $GITHUB_ENV
+          echo "${{ github.workspace }}/bin" >> $GITHUB_PATH
```

Comment:
I might be missing something obvious, but isn't this already done as part of checkout job and would not it be better to have one job for this test case because right now we have two separate and checkout job is not really interesting to PR author.

Reply:
> I might be missing something obvious, but isn't this already done as part of checkout job and would not it be better to have one job for this test case because right now we have two separate and checkout job is not really interesting to PR author.

As a disclaimer, I am mostly a total n00b with github workflows, since I always mostly write them by a 'monkey see monkey do' method... but I don't think jobs work like that, or at least that is not my experience. I started with this workflow like that and had to add the checkout and env. setup steps to all jobs, otherwise they simply couldn't find the sources. But maybe I am just doing it wrong.

Reply:
BTW, you can easily test if you can get it working without so much repetition. Just clone this branch, remove extra bits, then push to your own repo under any branch name starting with `test/`. That will trigger the action. Then you can check/follow how it goes from your repo's `Actions` tab.

Reply:
yep, I will test it and update you soon

Reply:
Well, of course that works. It has only 1 job.

> would not it be better to have one job for this test case because right now we have two separate and checkout job is not really interesting to PR author.

Okay, not that I really read your comment I notice I was way too hasty and failed to notice you suggested a single job. I thought you knew of a way to reuse the checked out repo in other jobs which depend (via `needs`) on `Checkout`. But I think/looks like it can't be done.

Anyway, what I really wanted to do is to readily set up the workflow so that
- there is a clear separation (jobs) between minimal PR verification and other code checks
- the minimal verification (eventually) checks sign-offs, formatting, 'go mod tidy'-ness, etc.
- other jobs depend on the success of minimal verification (`needs: [ verify ]`, only started if minimal checks pass)
- everything else is as much parallel as possible (for instance, one job per plugin e2e tests)

And you are right that checkout does not make any sense in itself. And actually verify as you suggest is probably the smallest set of minimal verification we can start with. Thanks !

Reply:
No worries and it is my bad that I was not clear enough on my comment above. I had an assumption that using needs should help us repeating the some steps but apparently every job runs on a separate environment which makes it impossible for use to use for example checked out code from one job in another job where we are running some tests against that checked out repo.